hi-ml-cpath/src/health_cpath/utils/montage.py

```diff
@@ -0,0 +1,517 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+
+import functools
+import logging
+import multiprocessing
+import shutil
+import sys
+import tempfile
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import torchvision
+import torch
+import numpy as np
+import pandas as pd
+from PIL import Image, ImageDraw, ImageFont
+from tqdm import tqdm
+from monai.data.image_reader import WSIReader
+
+from health_azure.utils import apply_overrides, parse_arguments
+from health_cpath.utils.montage_config import MontageConfig, create_montage_argparser
+from health_cpath.utils.naming import SlideKey
+from health_cpath.datasets.base_dataset import SlidesDataset
+from health_ml.utils.type_annotations import TupleInt3
+
+
+MONTAGE_FILE = "montage.png"
+DatasetOrDataframe = Union[SlidesDataset, pd.DataFrame]
+DatasetRecord = Dict[SlideKey, Any]
+
+
+def add_text(image: Image, text: str, y: float = 0.9, color: TupleInt3 = (27, 77, 40), fontsize_step: int = 2) -> None:
+    """Add text to a PIL image.
+
+    :param image: Image object to which text needs to be added.
+    :param text: The text that needs to be added.
+    :param y: Float between 0-1 specifying the vertical position of the text (default=0.9).
+    :param color: A 3-tuple indicating the fill color of the text (default = (27, 77, 40)).
+    :param fontsize_step: Steps of font size to reduce if the text size is more than image size (default=2).
+    """
+    # This fond is usually found in a path like /usr/share/fonts/truetype/dejavu
+    font_path = Path('DejaVuSans.ttf')
```

Comment:
from my recollection, this font was not found on Azure, should we add the font file in the same folder as the script?

Reply:
That's puzzling. I now added a smoke test for montage creation, it runs just fine. FYI, the code on cpath refenced a font file, but that was not in the repo either?
I'll resolve that, it does not appear to be an issue.